clang/lib/Analysis/ThreadSafety.cpp

```diff
@@ -1831,15 +1852,15 @@ void BuildLockset::handleCall(const Expr *Exp, const NamedDecl *D,
     assert(!Self);
     const auto *TagT = Exp->getType()->getAs<TagType>();
     if (D->hasAttrs() && TagT && Exp->isPRValue()) {
-      std::pair<til::LiteralPtr *, StringRef> Placeholder =
-          Analyzer->SxBuilder.createThisPlaceholder(Exp);
+      auto Placeholder = Analyzer->SxBuilder.createThisPlaceholder(Exp);
```

Comment:
I'd like to keep an explicit type here for readability. As an alternative, use the typed `std::get`, i.e. `std::get<StringRef>`.

Reply:
Growing this pair/tuple become a little too complex, so I just made createThisPlaceholder return a CapabilityExpr struct right away.